tests/handlers/test_directory.py

```diff
@@ -93,6 +94,33 @@ def test_get_remote_association(self):
             ignore_backoff=True,
         )
 
+    @defer.inlineCallbacks
```

Comment:
ahh, we need to talk about old-style async tests (which return a deferred) versus new-style synchronous tests which use a dummy Twisted reactor for the homeserver. In general the latter inherit from `tests.unittest.HomeserverTestCase`, though that's not essential, it just takes care of a bunch of the boilerplate of setting up a HomeServer using a dummy reactor, as well as providing a bunch of utility functions (including `get_success`, which will wait for a deferred to complete).

Since this test class is relatively small, I'd be inclined to convert it to new-style while you're here.

Reply:
See #6919.